[DOCS] Apply Readme comments

tsavina opened the pull request:
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

emmacall commented:
the three downloads banner is confusing, since you don't differentiate between pip / conda / etc in the labels

emmacall commented:
Is it confusing to add Geti as a tool since Geti is training focused and we try to clarify OV is inference?

emmacall commented:
Can we highlight more of the integrations? 
langchain, modelscope, easyocr, torchserve, sparknlp, etc.

rkazants commented on README.md:
```diff
@@ -38,9 +46,11 @@ Learn how to optimize and deploy popular models with the [OpenVINO Notebooks](ht
 Here are easy-to-follow code examples demonstrating how to run PyTorch and TensorFlow model inference using OpenVINO:
 
-**PyTorch Model**
+PyTorch Model
```
do we need details as for TensorFlow?

rkazants commented on README.md:
````diff
@@ -62,7 +72,8 @@ compiled_model = core.compile_model(ov_model, 'CPU')
 ```
 
-**TensorFlow Model**
+<details>
````
what is a reason to hide? I think these two chunks of code for TF and PT are the most important in this README to which developers will take a look

tsavina replied:
yeah agree
I just tested some options to save on space if we ever want to add more/llms examples, added back

rkazants commented on README.md:
```diff
@@ -83,26 +94,37 @@ compiled_model = core.compile_model(ov_model, 'CPU')
+
+[User documentation](https://docs.openvino.ai/) contains detailed information about OpenVINO and guides you from installation through optimizing and deploying models for your AI applications.
 
-OpenVINO also supports CPU, GPU, and NPU devices and works with models in TensorFlow, PyTorch, ONNX, TensorFlow Lite, PaddlePaddle model formats.
-With OpenVINO you can do automatic performance enhancements at runtime customized to your hardware (preserving model accuracy), including:
-asynchronous execution, batch processing, tensor fusion, load balancing, dynamic inference parallelism, automatic BF16 conversion, and more.
```
Why do we remove to mention all FWs? Strongly disagree.

tsavina replied:
to remove the fluff after it :) added links to devices,formats and APIs instead
pls check

rkazants reviewed (changes requested):
Why TF code is hidden? COde is the most important stuff to have a quick start to launch a model. Other part of this documentation is just words that users will not deep dive.
I am leaning to have rather short code samples here to demonstrate effective inference for different FWs, cases.

openvino-dev-samples commented on README.md:
```diff
@@ -83,26 +94,37 @@ compiled_model = core.compile_model(ov_model, 'CPU')
+-   [🤗Optimum Intel](https://github.com/huggingface/optimum-intel) - grab and use models leveraging OpenVINO within the Hugging Face API.
+-   [Torch.compile with OpenVINO](https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html) - use OpenVINO for Python-native applications by JIT-compiling code into optimized kernels.
+-   [OpenVINO LLMs with LlamaIndex](https://docs.llamaindex.ai/en/stable/examples/llm/openvino/) - build context-augmented GenAI applications with the LlamaIndex framework and enhance runtime performance with OpenVINO.
+-   [OpenVINO Execution Provider for ONNX Runtime]() - use OpenVINO as a backend with your existing ONNX Runtime code.
```
how about vLLM integration ?

tsavina replied:
thanks added!

openvino-dev-samples commented on README.md:
```diff
@@ -10,14 +18,14 @@
+Check out the [OpenVINO Cheat Sheet](https://docs.openvino.ai/2024/_static/download/OpenVINO_Quick_Start_Guide.pdf) and [Key Features](https://docs.openvino.ai/2024/about-openvino/key-features.html) for a quick reference.
+
 
 ## Installation
```
Shall mention GenAI API as one the option for LLM developer?

tsavina replied:
added in the APIs section

slyalin commented on README.md:
```diff
@@ -10,14 +18,14 @@
-Welcome to OpenVINO™, an open-source software toolkit for optimizing and deploying deep learning models.
 
 - **Inference Optimization**: Boost deep learning performance in computer vision, automatic speech recognition, generative AI, natural language processing with large and small language models, and many other common tasks.
 - **Flexible Model Support**: Use models trained with popular frameworks such as TensorFlow, PyTorch, ONNX, Keras, and PaddlePaddle. Convert and deploy models without original frameworks.
```
Please explicitly mention capability to take models built with `transformers` and `diffusers` from HuggingFace Hub with the help of optimum-intel. IMO, it is a more eye catching feature than just mentioning DL frameworks.

tsavina replied:
thanks, added!

tsavina commented:
@emmacall hi :)
> the three downloads banner is confusing, since you don't differentiate between pip / conda / etc in the labels
unfortunately we can't control the banners' text, best I can do is to align it in the table which looks quite bad.

> Can we highlight the path to the genai readme more somehow?
what about a small separate section for genai install and tutorials, added a couple of small paragraphs, let me know if this is extra  

> Is it confusing to add Geti as a tool since Geti is training focused and we try to clarify OV is inference?
We need to decide whether we consider Geti as OV ecosystem or not 

> Can we highlight more of the integrations? langchain, modelscope, easyocr, torchserve, sparknlp, etc.
added some of them, I don't think the long list of integrations in the readme makes sense, what do you think about highlighting the most important ones and giving a link to docs (not published yet) or hiding the rest in the collapsible

openvino-dev-samples commented on README.md:
```diff
@@ -84,25 +94,52 @@ data = np.random.rand(1, 224, 224, 3)
-## Documentation
+-   [🤗Optimum Intel](https://github.com/huggingface/optimum-intel) - grab and use models leveraging OpenVINO within the Hugging Face API.
+-   [Torch.compile](https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html) - use OpenVINO for Python-native applications by JIT-compiling code into optimized kernels.
+-   [OpenVINO LLMs inference and serving with vLMM​](https://docs.vllm.ai/en/stable/getting_started/openvino-installation.html) - enhance vLLM's fast and easy model serving with the OpenVINO backend.
```
here is a typo "vLMM" 😀

openvino-dev-samples replied:
```suggestion
-   [OpenVINO LLMs inference and serving with vLLM​](https://docs.vllm.ai/en/stable/getting_started/openvino-installation.html) - enhance vLLM's fast and easy model serving with the OpenVINO backend.
```

kblaszczak-intel reviewed (approved):
Since this one has been hanging here for a while, I think we should merge and continue polishing in a new PR, if necessary. 
It's better to have some improvements published early than to wait months for a "perfect" version.
@rkazants Does it look acceptable?

kblaszczak-intel commented:
Guys, a quick reminder ;)  Can we merge this one? Can I do something to help here?

rkazants commented on README.md:
```diff
@@ -10,14 +18,14 @@
 
 - **Inference Optimization**: Boost deep learning performance in computer vision, automatic speech recognition, generative AI, natural language processing with large and small language models, and many other common tasks.
-- **Flexible Model Support**: Use models trained with popular frameworks such as TensorFlow, PyTorch, ONNX, Keras, and PaddlePaddle. Convert and deploy models without original frameworks.
+- **Flexible Model Support**: Use models trained with popular frameworks such as PyTorch, TensorFlow, ONNX, Keras, and PaddlePaddle. Directly integrate models built with transformers and diffusers from the Hugging Face Hub using Optimum Intel. Convert and deploy models without original frameworks.
```
Let us have a link to https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html
when we say "Convert and deploy models without original frameworks"

rkazants commented on README.md:
```diff
@@ -86,25 +96,43 @@ data = np.random.rand(1, 224, 224, 3)
-OpenVINO also supports CPU, GPU, and NPU devices and works with models in TensorFlow, PyTorch, ONNX, TensorFlow Lite, PaddlePaddle model formats.
-With OpenVINO you can do automatic performance enhancements at runtime customized to your hardware (preserving model accuracy), including:
-asynchronous execution, batch processing, tensor fusion, load balancing, dynamic inference parallelism, automatic BF16 conversion, and more.
+OpenVINO supports the CPU, GPU, and NPU [devices](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes.html) and works with models from PyTorch, TensorFlow, ONNX, TensorFlow Lite, and PaddlePaddle [frameworks](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html). It includes [APIs](https://docs.openvino.ai/2024/api/api_reference.html) in ...
```
```suggestion
OpenVINO supports the CPU, GPU, and NPU [devices](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes.html) and works with models from PyTorch, TensorFlow, ONNX, TensorFlow Lite, PaddlePaddle, and JAX/Flax [frameworks](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html). It includes [APIs](https://docs.openvino.ai/2024/api/api_reference.html) in C++, Python, C, NodeJS, and offers the GenAI API for optimized model pipelines and performance.
```

kblaszczak-intel commented on README.md:
```diff
@@ -86,25 +96,43 @@ data = np.random.rand(1, 224, 224, 3)
-OpenVINO also supports CPU, GPU, and NPU devices and works with models in TensorFlow, PyTorch, ONNX, TensorFlow Lite, PaddlePaddle model formats.
-With OpenVINO you can do automatic performance enhancements at runtime customized to your hardware (preserving model accuracy), including:
-asynchronous execution, batch processing, tensor fusion, load balancing, dynamic inference parallelism, automatic BF16 conversion, and more.
+OpenVINO supports the CPU, GPU, and NPU [devices](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes.html) and works with models from PyTorch, TensorFlow, ONNX, TensorFlow Lite, PaddlePaddle, and JAX/Flax [frameworks](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html). It includes [APIs](https://docs.openvino.ai/2024/api/api_referenc ...
```
```suggestion
OpenVINO supports the CPU, GPU, and NPU [devices](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes.html) and works with models from PyTorch, TensorFlow, ONNX, TensorFlow Lite, PaddlePaddle, and JAX/Flax [frameworks](https://docs.openvino.ai/2024/openvino-workflow/model-preparation.html). It includes [APIs](https://docs.openvino.ai/2024/api/api_reference.html) in C++, Python, C, and NodeJS.
```

kblaszczak-intel commented on README.md:
```diff
@@ -86,25 +96,43 @@ data = np.random.rand(1, 224, 224, 3)
+
+## Generative AI with OpenVINO
+
+Get started with the OpenVINO GenAI [installation](https://docs.openvino.ai/2024/get-started/install-openvino/install-openvino-genai.html) and refer to the [detailed guide](https://docs.openvino.ai/2024/learn-openvino/llm_inference_guide/genai-guide.html) to explore the capabilities of Generative AI using OpenVINO.
```
```suggestion
For optimized Generative AI model pipelines, [install OpenVINO GenAI](https://docs.openvino.ai/2024/get-started/install-openvino/install-openvino-genai.html) and refer to the [detailed guide](https://docs.openvino.ai/2024/learn-openvino/llm_inference_guide/genai-guide.html).
```

kblaszczak-intel commented:
Hey everyone, can we finally merge this one? It doesn't need to be perfect, if there are any concerns, let's address them in the next revision, please.

rkazants commented:
> Hey everyone, can we finally merge this one? It doesn't need to be perfect, if there are any concerns, let's address them in the next revision, please.

yes, let us merge it
